test/framework/cluster.go

```diff
@@ -830,22 +862,58 @@ func (e *ClusterE2ETest) InstallHelmChart() {
 	kubeconfig := e.kubeconfigFilePath()
 	ctx := context.Background()
 
-	err := e.HelmInstallConfig.HelmClient.InstallChart(ctx, e.HelmInstallConfig.chartName, e.HelmInstallConfig.chartURI, e.HelmInstallConfig.chartVersion, kubeconfig, e.HelmInstallConfig.chartValues)
+	err := e.HelmInstallConfig.HelmClient.InstallChart(
+		ctx,
+		e.HelmInstallConfig.chartName,
+		e.HelmInstallConfig.chartURI,
+		e.HelmInstallConfig.chartVersion,
+		kubeconfig,
+		e.HelmInstallConfig.chartValues,
+	)
 	if err != nil {
 		e.T.Fatalf("Error installing %s helm chart on the cluster: %v", e.HelmInstallConfig.chartName, err)
 	}
 }
 
+func (e *ClusterE2ETest) CreateNamespace(namespace string) {
+	kubeconfig := e.kubeconfigFilePath()
+	err := e.KubectlClient.CreateNamespace(context.Background(), kubeconfig, namespace)
+	if err != nil {
+		e.T.Fatalf("Namespace creation failed for %s", namespace)
+	}
+}
+
+func (e *ClusterE2ETest) DeleteNamespace(namespace string) {
+	kubeconfig := e.kubeconfigFilePath()
+	err := e.KubectlClient.DeleteNamespace(context.Background(), kubeconfig, namespace)
+	if err != nil {
+		e.T.Fatalf("Namespace deletion failed for %s", namespace)
+	}
+}
+
 func (e *ClusterE2ETest) InstallCuratedPackagesController() {
 	kubeconfig := e.kubeconfigFilePath()
 	// TODO Add a test that installs the controller via the CLI.
 	ctx := context.Background()
-	err := e.PackageConfig.HelmClient.InstallChart(ctx,
-		e.PackageConfig.chartName, e.PackageConfig.chartURI,
-		e.PackageConfig.chartVersion, kubeconfig, e.PackageConfig.chartValues)
+	charts, err := e.PackageConfig.HelmClient.ListCharts(ctx, kubeconfig)
 	if err != nil {
-		e.T.Fatalf("Error installing %s helm chart on the cluster: %v",
-			e.PackageConfig.chartName, err)
+		e.T.Fatalf("Unable to list charts: %v", err)
```

Comment:
This is better handled with two separate functions, IMO:

IsCuratedPackagesControllerInstalled and InstallCuratedPackagesController

The behavior of this function doesn't align well with its name. Having two functions is simpler, and reads more obviously.